### DISCUSSION TITLE
How to setup C++ program's indent as 4

### DISCUSSION THREAD

**Discussion by dlintw** [LazyVim/LazyVim#1500] [Q&A] [answered]
I've tried to figure out the nulls's document, but failed at the first step Debug.

In null's document, https://github.com/jose-elias-alvarez/null-ls.nvim#how-do-i-enable-debug-mode-and-get-debug-output
```
require("null-ls").setup({
    debug = true,
})
```
I try to enable debug. In my ~/.config/nvim/lua/plugin/nullls.lua, this setting can't work.
```
return {
	{
		"jose-elias-alvarez/null-ls.nvim",
		debug = true,
		opts = function(_, opts)
			local nls = require("null-ls")
			opts.sources = vim.list_extend(opts.sources, {
				nls.builtins.formatting.shfmt.with({
					extra_args = { "-i", "2", "-ci", "-bn" },
				}),
			})
		end,
	},
}
```
edit a sample.cpp
```
#include <stdio.h>
int main() {
}
```
Use :LspInfo the result is
```
Language client log: /home/hosthome/.local/state/
Detected filetype:   cpp

0 client(s) attached to this buffer:

Configured servers list: tsserver, lua_ls, jsonls
```

**Comment by dpetka2001** [comment, marked as answer]
Create the following path if the directory and file don't exist: `~/.config/nvim/after/ftplugin/cpp.lua` with the following contents
```lua
vim.opt_local.tabstop = 4
vim.opt_local.shiftwidth = 4
vim.opt_local.expandtab = true
```
